packages/core/src/utils/transformKeys.ts

```diff
@@ -0,0 +1,38 @@
+import { outdent } from './outdent';
+
+class IllegalFieldNameError extends Error {
+  constructor(illegalSequence: string) {
+    super();
+    this.message = outdent`
+      The sequence "${illegalSequence}" is reserved and not allowed in field names
+      Either:
+        - remove all instances of "${illegalSequence}" in the names of fields in your content
+        - add a fieldTransform function to your flatbread.config.js to translate to something else
+          Example:
+            {
+              ...,
+              fieldTransform: (value) => value.replaceAll("${illegalSequence}",'-')
+            }
+      `;
+  }
+}
+
+function isObject(obj: any): obj is Object {
+  return obj != null && obj.constructor.name === 'Object';
+}
+
+export default function transformKeys(
+  obj: any,
+  transform: (key: string) => string
+): any {
```

Comment:
can we narrow these `any` types?

Reply:
any ideas how to do this?  it's a recursive function with the top level call being obj: EntryNode

Reply:
looks like unknown works